test/e2e/cluster_create_missing_info.go

```diff
@@ -46,14 +46,10 @@ var _ = Describe("Customer", func() {
 					customerVnetSubnetName           = "customer-vnet-subnet1"
 					customerClusterName              = "illegal-hcp-cluster"
 				)
-				ic := framework.InvocationContext()
+				ic := framework.NewInvocationContext()
 
 				By("creating a resource group")
-				resourceGroup, cleanupResourceGroup, err := ic.NewResourceGroup(ctx, "basic-create", "uksouth")
-				DeferCleanup(func(ctx SpecContext) {
```

Comment:
so now we don't have to remember to call a defer cleanup.  it just happens along with (soon) collection of debug info for failed tests.